arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -1070,250 +1373,254 @@ bool fromFuncExists(
+        // any string
+        {
+          irs::string_ref("string"),
+          [] (std::string& name,
+              IResearchAnalyzerFeature::AnalyzerPool const&) {
+            kludge::mangleAnalyzer(name);
+            return true; // a prefix match
+          }
+        },
+        // any non-string type
+        {
+          irs::string_ref("type"),
+          [](std::string& name,
+             IResearchAnalyzerFeature::AnalyzerPool const&) {
+            kludge::mangleType(name);
+            return true; // a prefix match
+          }
+        },
+        // concrete analyzer from the context
+        {
+          TypeAnalyzer,
+          [](std::string& name,
+             IResearchAnalyzerFeature::AnalyzerPool const& analyzer) {
+            kludge::mangleStringField(name, analyzer);
+            return false; // not a prefix match
+          }
+        },
+        {
+          irs::string_ref("numeric"),
+          [] (std::string& name, IResearchAnalyzerFeature::AnalyzerPool const&) {
+            kludge::mangleNumeric(name);
+            return false; // not a prefix match
+          }
+        },
+        {
+          irs::string_ref("bool"),
+          [] (std::string& name, IResearchAnalyzerFeature::AnalyzerPool const&) {
+            kludge::mangleBool(name);
+            return false; // not a prefix match
+          }
+        },
+        {
+          irs::string_ref("boolean"),
+          [] (std::string& name, IResearchAnalyzerFeature::AnalyzerPool const&) {
+            kludge::mangleBool(name);
+            return false; // not a prefix match
+          }
+        },
+        {
+          irs::string_ref("null"),
+          [] (std::string& name, IResearchAnalyzerFeature::AnalyzerPool const&) {
+            kludge::mangleNull(name);
+            return false; // not a prefix match
+          }
+        }
+      };
 
-      // 3rd argument defines a value (if present)
-      auto const analyzerArg= args.getMemberUnchecked(2);
+      auto const typeHandler = TypeHandlers.find(strArg);
 
-      if (!analyzerArg) {
+      if (TypeHandlers.end() == typeHandler) {
         LOG_TOPIC(WARN, arangodb::iresearch::TOPIC)
-          << "'EXISTS' AQL function: 3rd argument is invalid";
+            << "'EXISTS' AQL function: 2nd argument must be equal to one of the following:"
+               " 'string', 'type', 'analyzer', 'numeric', 'bool', 'boolean', 'null', but got '" << arg << "'";
+        return false;
       }
 
-      arangodb::iresearch::ScopedAqlValue analyzerId(*analyzerArg);
-
-      if (filter || analyzerId.isConstant()) {
-
-        if (!analyzerId.execute(ctx)) {
-          LOG_TOPIC(WARN, arangodb::iresearch::TOPIC)
-            << "'EXISTS' AQL function: Failed to evaluate 3rd argument";
-          return false;
-        }
-
-        if (!analyzerId.getString(arg)) {
+      if (argc > 2) {
+        if (TypeAnalyzer.c_str() != typeHandler->first.c_str()) {
```

Comment:
is this intended as a string comparion?
if not, can we make sure the string pointers will always be the same?

Reply:
that is intentional pointer comparison.
`irs::string_ref const TypeAnalyzer("analyzer");` is defined above and will contain same `const char*`

Reply:
ok